plugin/rewrite/cname_target.go

```diff
@@ -0,0 +1,161 @@
+package rewrite
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+// UpstreamInt wraps the Upstream API for dependency injection during testing
+type UpstreamInt interface {
+	Lookup(ctx context.Context, state request.Request, name string, typ uint16) (*dns.Msg, error)
+}
+
+// RewriteType is the type of cname target rewrite rules
+type RewriteType string
+
+const (
+	// CNameExactMatch is the type for exact match of cname target
+	CNameExactMatch RewriteType = "exact"
+	// CNamePrefixMatch is the type for prefix match of cname target
+	CNamePrefixMatch RewriteType = "prefix"
+	// CNameSuffixMatch is the type for suffix match of cname target
+	CNameSuffixMatch RewriteType = "suffix"
+	// CNameSubstringMatch is the type for substring match of cname target
+	CNameSubstringMatch RewriteType = "substring"
+	// CNameRegexMatch is the type for regex match of cname target
+	CNameRegexMatch RewriteType = "regex"
+)
```

Comment:
@Amila-Rukshan , why do you need to add the above when we already have the following?

https://github.com/coredns/coredns/blob/0063d7a80c0db18069429c775e4b95a5c0b4b69c/plugin/rewrite/name.go#L115-L124

Reply:
Updated the code to reuse the rewrite types